code/modules/research/bepis.dm

```diff
@@ -0,0 +1,231 @@
+		say("ERROR: CRITICAL MACHIME MALFUNCTI- ON. CURRENCY IS NOT CRASH. CANNOT COMPUTE COMMAND: 'make bucks'") //not a typo, for once.
+		new /mob/living/simple_animal/deer(dropturf, 1)
+		use_power(MACHINE_OVERLOAD * power_saver) //To prevent gambling at low cost and also prevent spamming for infinite deer.
+		return
+	//Minor Failure
+	error_cause = pick("attempted to sell grey products to American dominated market.","attempted to sell gray products to British dominated market.","placed wild assumption that PDAs would go out of style.","simulated product #76 damaged brand reputation mortally.","simulated business model resembled 'pyramid scheme' by 98.7%.","product accidently granted override access to all station doors.")
+	say("Experiment concluded with zero product viability. Cause of error: [error_cause]")
+	return
+
+/obj/machinery/rnd/bepis/update_icon_state()
+	if(panel_open == TRUE)
+		icon_state = "chamber_open"
+		return
+	if((powered == TRUE) && is_operational())
+		icon_state = "chamber_active"
+		return
+	if((powered == TRUE) && (banked_cash > 0) && (is_operational()))
+		icon_state = "chamber_active_loaded"
+		return
+	if (((powered == FALSE) && (banked_cash > 0)) || (banked_cash > 0) && (!is_operational()))
+		icon_state = "chamber_loaded"
+		return
+	if(((powered == FALSE) && (banked_cash == 0)) || (!is_operational()))
+		icon_state = "chamber"
+		return
+
+/obj/machinery/rnd/bepis/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "bepis", name, 500, 480, master_ui, state)
+		ui.open()
+	RefreshParts()
+
+/obj/machinery/rnd/bepis/ui_data(mob/user)
+	var/list/data = list()
+	data["account_owner"] = account_name
+	data["amount"] = banking_amount
+	data["stored_cash"] = banked_cash
+	data["mean_value"] = major_threshold
+	data["error_name"] = error_cause
+	data["power_saver"] = power_saver
+	data["accuracy_percentage"] = inaccuracy_percentage * 100
+	data["positive_cash_offset"] = positive_cash_offset
+	data["negative_cash_offset"] = negative_cash_offset
+	data["manual_power"] = powered ? FALSE : TRUE
+	data["silicon_check"] = issilicon(user)
+	return data
+
+/obj/machinery/rnd/bepis/ui_act(action,params)
+	if(..())
+		return
+	switch(action)
+		if("deposit_cash")
+			if(powered == FALSE)
+				return
+			depositcash()
+		if("withdraw_cash")
+			if(powered == FALSE)
+				return
+			withdrawcash()
+		if("begin_experiment")
+			if(powered == FALSE)
+				return
+			if(banked_cash == 0)
+				say("Please deposit funds to begin testing.")
+				return
+			calcsuccess()
+			use_power(MACHINE_OPERATION * power_saver) //This thing should eat your APC battery if you're not careful.
+			powered = FALSE	//Shuts off after use in order to get the right visual look. Kinda annoying, but also prevents some possible spam issues.
```

Comment:
Should happen when you replace powered, but make sure you actually update idle/active here.